### DISCUSSION TITLE
How to use main, stream, command in Raspberry PI4 not just cli?

### DISCUSSION THREAD

**Discussion by ValeriyDP** [ggml-org/whisper.cpp#1215] [Q&A]
Hi!

I'm just wondering how to use any of the binaries (main, stream, command) in Python code. I mean that looks like I have to create a service and run it background, but how to get output then? Or my code should run it using some trigger and launch any of those binaries in foreground, but there is the collision that trigger is a voice command... Could someone give me a hand, please?

Is that https://github.com/abdeladim-s/pywhispercpp good place to start?

Thank you in advance.

**Comment by SiddiqMohammed** [comment]
Hello @ValeriyDP , were you able to figure it out?

I've been looking for similar options and found pywhisper but wasn't able to get it working with my raspberry pi 4. Do let me know if you were able to make any progress with this! 

Thanks.
